statsmodels/sandbox/regression/gmm.py

```diff
@@ -70,38 +73,36 @@ def maxabs(x):
 
 
 class IV2SLS(LikelihoodModel):
-    '''
-    Class for instrumental variables estimation using Two-Stage Least-Squares
+    """
+    Instrumental variables estimation using Two-Stage Least-Squares (2SLS)
 
 
     Parameters
     ----------
-    endog: array 1d
-       endogenous variable
+    endog: array
+       Endogenous variable, 1-dimensional or 2-dimensional array nobs by 1
     exog : array
-       explanatory variables
+       Explanatory variables, 1-dimensional or 2-dimensional array nobs by k
     instruments : array
-       instruments for explanatory variables, needs to contain those exog
-       variables that are not instrumented out
+       Instruments for explanatory variables. Must contain both exog
+       variables that are not being instrumented and instruments
 
     Notes
     -----
     All variables in exog are instrumented in the calculations. If variables
-    in exog are not supposed to be instrumented out, then these variables
-    need also to be included in the instrument array.
+    in exog are not supposed to be instrumented, then these variables
+    must also to be included in the instrument array.
 
     Degrees of freedom in the calculation of the standard errors uses
     `df_resid = (nobs - k_vars)`.
     (This corresponds to the `small` option in Stata's ivreg2.)
-
-
-    '''
+    """
 
     def __init__(self, endog, exog, instrument=None):
-        self.instrument = instrument
+        self.instrument = _ensure_2d(instrument)
```

Comment:
Pandas works fine for all 3 inputs.



On Sun, Jun 19, 2016 at 6:45 PM Josef Perktold <notifications@github.com>
wrote:

> In statsmodels/sandbox/regression/gmm.py
> <https://github.com/statsmodels/statsmodels/pull/3040#discussion_r67620949>
> :
>
> >
> >      def __init__(self, endog, exog, instrument=None):
> > -        self.instrument = instrument
> > +        self.instrument = _ensure_2d(instrument)
>
> instrument *must* be a plain ndarray internally.
>
> We don't want to have to pay attention how other data types might behave
> inside models.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/statsmodels/statsmodels/pull/3040/files/2f3809457f42e479246ac7f30dd18ffcfb877e54#r67620949>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFU5RcGvWXgqzHOTvUEHnLu7LE9RanREks5qNYA0gaJpZM4I5NBI>
> .
>